web-js/src/components/RealmApp.js

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import * as Realm from "realm-web";
+
+const RealmAppContext = React.createContext(null);
+
+export function RealmAppProvider({ appId, children }) {
+  // Store Realm.App in React state. If appId changes, all children will rerender and use the new realmApp.
+  const [realmApp, setRealmApp] = React.useState(new Realm.App(appId));
+  React.useEffect(() => {
+    setRealmApp(new Realm.App(appId));
+  }, [appId]);
+  // Store the app's current user in state and wrap the built-in auth functions to modify this state
+  const [currentUser, setCurrentUser] = React.useState(realmApp.currentUser);
+  // Wrap the base logIn function to save the logged in user in state
+  const logIn = React.useCallback(
+    async (credentials) => {
+      await realmApp.logIn(credentials);
+      setCurrentUser(realmApp.currentUser);
+    },
+    [realmApp]
+  );
+  // Wrap the current user's logOut function to remove the logged out user from state
+  const logOut = React.useCallback(async () => {
+    await realmApp.currentUser?.logOut();
```

Comment:
Should we remove the user by calling `Realm.App.removeUser()`?

Reply:
Not strictly necessary, but we have no reason to hold on to the user's data and it demonstrates another method so sure thing!